### PR TITLE
Update packages

### PR DESCRIPTION
**Pull request by gdalle**
Big update on the webpage to make it more accessible to newcomers:

- Revamp the introduction to present all differentation paradigms
- Remove most technicalities and add links to expository papers instead
- Trim package list by putting inactive ones at the bottom
- Add new packages that were not present
- Add whole new section on differentiating through solvers
- Add my name to the author list (if that's okay)

### CODE REVIEW

**Comment by gdalle** [comment]
Here's a preview of the result
[JuliaDiff.pdf](https://github.com/JuliaDiff/juliadiff.github.io/files/14095053/JuliaDiff.pdf)

File: index.md [hunk 1]
```diff
@@ -3,87 +3,115 @@
...
 
 ## The Big List
-This is a big list of Julia Automatic Differentiation (AD) packages and related tooling.
-As you can see there is a lot going on here.
-As with any such big lists it rapidly becomes out-dated.
-When you notice something that is out of date, or just plain wrong, please [submit a PR](https://github.com/JuliaDiff/juliadiff.github.io).
 
-This list aims to be comprehensive in coverage.
+What follows is a big list of Julia differentiation packages and related tooling, last updated in January 2024.
+If you notice something inaccurate or outdated, please [open an issue](https://github.com/JuliaDiff/juliadiff.github.io/issues) to signal it.
```
Comment (oxinabox) [inline index.md:37]: Why the change from submit a PR to open an issue?

File: index.md [hunk 2]
```diff
@@ -3,87 +3,115 @@
...
-- [SparseDiffTools.jl](https://github.com/JuliaDiff/SparseDiffTools.jl): Exploiting sparsity to speed up FiniteDiff.jl and ForwardDiff.jl, as well as other algorithms.
+
+These packages define derivatives for basic functions, and enable users to do the same:
+
+- [JuliaDiff/ChainRules](https://www.juliadiff.org/ChainRulesCore.jl/stable/): Ecosystem for backend-agnostic forward and reverse rules.
+  - [JuliaDiff/ChainRulesCore.jl](https://github.com/JuliaDiff/ChainRulesCore.jl): Core API for users to add rules to their package.
+  - [JuliaDiff/ChainRules.jl](https://github.com/JuliaDiff/ChainRules.jl/): Rules for Julia Base and standard libraries.
+  - [JuliaDiff/ChainRulesTestUtils.jl](https://github.com/JuliaDiff/ChainRulesTestUtils.jl/): Tools for testing rules defined with ChainRulesCore.jl.
+- [ThummeTo/ForwardDiffChainRules.jl](https://github.com/ThummeTo/ForwardDiffChainRules.jl): Translate rules from ChainRulesCore.jl to make them compatible with ForwardDiff.jl
+- [JuliaDiff/DiffRules.jl](https://github.com/JuliaDiff/DiffRules.jl): Scalar rules used by ForwardDiff.jl
```
Comment (devmotion) [inline index.md:82]:
```suggestion
- [JuliaDiff/DiffRules.jl](https://github.com/JuliaDiff/DiffRules.jl): Scalar rules used by ForwardDiff.jl, ReverseDiff.jl, and Tracker.jl
```

File: index.md [hunk 3]
```diff
@@ -3,87 +3,115 @@
...
-- [Calculus.jl](https://github.com/JuliaMath/Calculus.jl): Largely deprecated, legacy package. New users should look to FiniteDifferences.jl and FiniteDiff.jl instead.
+### Numeric differentiation
+
+- [JuliaDiff/FiniteDifferences.jl](https://github.com/JuliaDiff/FiniteDifferences.jl): Finite differences with support for arbitrary types and higher order schemes
+- [JuliaDiff/FiniteDiff.jl](https://github.com/JuliaDiff/FiniteDiff.jl): Finite differences with support for caching and sparsity
+
+### Higher order
+
+- [JuliaDiff/TaylorSeries.jl](https://github.com/JuliaDiff/TaylorSeries.jl): Taylor polynomial expansions in one or more variables
+- [JuliaDiff/TaylorDiff.jl](https://github.com/JuliaDiff/TaylorDiff.jl): Higher order directional derivatives (experimental)
```
Comment (oxinabox) [inline index.md:72]: We should put Diffractor here too.
Diffractor's forward mode is actually a Taylor forward mode.

File: index.md [hunk 4]
```diff
@@ -3,87 +3,115 @@
...
+
+- [JuliaDiff/SparseDiffTools.jl](https://github.com/JuliaDiff/SparseDiffTools.jl): Exploit sparsity to speed up FiniteDiff.jl and ForwardDiff.jl, as well as other algorithms.
+- [gaurav-arya/StochasticAD.jl](https://github.com/gaurav-arya/StochasticAD.jl): Differentiation of functions with stochastic behavior (experimental)
+
+### Differentiating through more stuff
+
+Some complex algorithms are not natively differentiable, which is why derivatives have been implemented in the following packages:
+
+- [SciML](https://github.com/SciML): For a lot of different domains of scientific machine learning: differential equations, linear and nonlinear systems, optimization problems, etc.
+- [TuringLang/DistributionsAD.jl](https://github.com/TuringLang/DistributionsAD.jl): For probability distributions
```
Comment (devmotion) [inline index.md:99]: Not sure if this should be listed, it's a type piracy mess and the intention is to move definitions and workarounds to upstream packages such as Distributions, ReverseDiff etc.
```suggestion
```

Comment (oxinabox) [inline index.md:99]:
```suggestion
- [TuringLang/DistributionsAD.jl](https://github.com/TuringLang/DistributionsAD.jl): Extensions to Distributions.jl to support AD. Somewhat obsoleted by these moving into Distributions.jl extensions.
```

Or maybe just delete it?

**Review by oxinabox — APPROVED** [review]

File: index.md (hunk 4 above)
Comment (gdalle) [reply index.md:99]: I would still mention it, if only to tell people exactly that

File: index.md (hunk 1 above)
Comment (gdalle) [reply index.md:37]: Because beginners will find it easier to open an issue than to fork, clone, modify, push and PR

File: index.md (hunk 4 above)
Comment (gdalle) [reply index.md]: I deleted it following @devmotion's remark

File: index.md (hunk 3 above)
Comment (gdalle) [reply index.md:72]: or we could put the Taylor stuff inside forward mode? conceptually it's the same, right?

File: index.md [hunk 5]
```diff
@@ -3,87 +3,114 @@
...
-- [SparseDiffTools.jl](https://github.com/JuliaDiff/SparseDiffTools.jl): Exploiting sparsity to speed up FiniteDiff.jl and ForwardDiff.jl, as well as other algorithms.
+
+These packages define derivatives for basic functions, and enable users to do the same:
+
+- [JuliaDiff/ChainRules](https://www.juliadiff.org/ChainRulesCore.jl/stable/): Ecosystem for backend-agnostic forward and reverse rules.
+  - [JuliaDiff/ChainRulesCore.jl](https://github.com/JuliaDiff/ChainRulesCore.jl): Core API for users to add rules to their package.
+  - [JuliaDiff/ChainRules.jl](https://github.com/JuliaDiff/ChainRules.jl/): Rules for Julia Base and standard libraries.
+  - [JuliaDiff/ChainRulesTestUtils.jl](https://github.com/JuliaDiff/ChainRulesTestUtils.jl/): Tools for testing rules defined with ChainRulesCore.jl.
+- [ThummeTo/ForwardDiffChainRules.jl](https://github.com/ThummeTo/ForwardDiffChainRules.jl): Translate rules from ChainRulesCore.jl to make them compatible with ForwardDiff.jl
+- [JuliaDiff/DiffRules.jl](https://github.com/JuliaDiff/DiffRules.jl): Scalar rules used by ForwardDiff.jl, ReverseDiff.jl, and Tracker.jl
```
Comment (devmotion) [inline index.md:82]: Actually, the list is much longer:
```suggestion
- [JuliaDiff/DiffRules.jl](https://github.com/JuliaDiff/DiffRules.jl): Scalar rules used by e.g. ForwardDiff.jl, ReverseDiff.jl, Zygote.jl, Tracker.jl, and Symbolics.jl

Comment (oxinabox) [reply index.md]: Zygote doesn't actually use DiffRules.jl except for its vestigial ForwardMode which noone uses.

File: index.md (hunk 3 above)
Comment (gdalle) [reply index.md:72]: I put Diffractor here

File: index.md (hunk 5 above)
Comment (gdalle) [reply index.md]: fixed

**Pull request merged**